command/connect/envoy/exec_windows.go

```diff
@@ -0,0 +1,121 @@
+//go:build windows
+// +build windows
+
+package envoy
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"path/filepath"
+	"strings"
+
+	"time"
+)
+
+// testSelfExecOverride is a way for the tests to no fork-bomb themselves by
+// self-executing the whole test suite for each case recursively. It's gross but
+// the least gross option I could think of.
+var testSelfExecOverride string
+
+func isHotRestartOption(s string) bool {
+	restartOpts := []string{
+		"--restart-epoch",
+		"--hot-restart-version",
+		"--drain-time-s",
+		"--parent-shutdown-time-s",
+	}
+	for _, opt := range restartOpts {
+		if s == opt {
+			return true
+		}
+		if strings.HasPrefix(s, opt+"=") {
+			return true
+		}
+	}
+	return false
+}
+
+func hasHotRestartOption(argSets ...[]string) bool {
+	for _, args := range argSets {
+		for _, opt := range args {
+			if isHotRestartOption(opt) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func makeBootstrapTemp(bootstrapJSON []byte) (string, error) {
+	tempFile := filepath.Join(os.TempDir(),
+		fmt.Sprintf("envoy-%x-bootstrap.json", time.Now().UnixNano()+int64(os.Getpid())))
+
+	f, err := os.Create(tempFile)
+	if err != nil {
+		return tempFile, err
+	}
+
+	defer f.Close()
+	f.Write(bootstrapJSON)
+	f.Sync()
+
+	// We can't wait for the process since we need to exec into Envoy before it
+	// will be able to complete so it will be remain as a zombie until Envoy is
+	// killed then will be reaped by the init process (pid 0). This is all a bit
+	// gross but the cleanest workaround I can think of for Envoy 1.10 not
+	// supporting /dev/fd/<fd> config paths any more. So we are done and leaving
+	// the child to run it's course without reaping it.
+	return tempFile, nil
+}
```

Comment:
This could be a security concerns. The temp file in here will potentially have the agent token written into it. Is there a way in Windows to emulate an in-memory file? maybe something like https://learn.microsoft.com/en-us/dotnet/api/system.io.memorymappedfiles.memorymappedfile.createnew?view=net-7.0 for golang?

Reply:
addressed in https://github.com/hashicorp/consul/pull/17694